src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,205 @@
+                                "Format: description /by time");
+                    }
+                }
+                String name = command.split(" /by")[0].trim();
+                String time = command.split(" /by")[1].trim();
+                if (isValidDate(time)) {
+                    LocalDate todoDate = LocalDate.parse(time);
+                    Deadline deadline = new Deadline(name, todoDate);
+                    output = taskList.addDeadline(deadline);
+                } else {
+                    Deadline deadline = new Deadline(name, time);
+                    output = taskList.addDeadline(deadline);
+                }
+            } else if (command.split(" ")[0].equals("event")) { // add event
+                command = command.substring(command.split(" ")[0].length() + 1, command.length()).trim();
+                if (command.split("/at").length == 0) {
+                    throw new InvalidEventException("OOPS! The description and time is missing. " +
+                            "Format: description /at time");
+                } else if (command.split("/at").length == 1) {
+                    if (command.split("/at")[0].equals("")) {
+                        throw new InvalidEventException("OOPS! The description and time is missing. " +
+                                "Format: description /at time");
+                    } else {
+                        throw new InvalidEventException("OOPS! The time is missing. " +
+                                "Format: description /at time");
+                    }
+                } else if (command.split("/at").length == 2) {
+                    if (command.split("/at")[0].equals("")) {
+                        throw new InvalidEventException("OOPS! The description is missing.");
+                    }
+                }
+                String name = command.split(" /at")[0].trim();
+                String time = command.split(" /at")[1].trim();
+                if (isValidDate(time)) {
+                    LocalDate todoDate = LocalDate.parse(time);
+                    Event event = new Event(name, todoDate);
+                    output = taskList.addEvent(event);
+                } else {
+                    Event event = new Event(name, time);
+                    output = taskList.addEvent(event);
+                }
+            } else {
+                throw new InvalidInputException();
+            }
+        return output;
+        }
+
+    /**
+     * Process tasks from the database to a list.
+     * @param line Every line of task from the database.
+     * @param storedTasks The list of task to load with.
+     */
+    public void processInputFromFile(String line, List<Task> storedTasks) {
+        String[] split = line.split("\\ \\|\\ ");
+        String type = split[0];
+        switch (type) {
+            case "T":
+                Task task = new Task(split[2]);
```

Comment:
would it be helpful to extract individual parse functions to separate methods to make it easier to debug?